### ISSUE TITLE
Autocomplete input with prefilled v-model causes unnecessary call to options method

### ISSUE THREAD

**Issue by mweghorst** [formkit/formkit#657]
### Reproduction

https://formkit.link/09482f6f5095df7c99617c4f1039f742

### Describe the bug

When using a FormKit autocomplete input with v-model, the options loader is called whenever an option is selected. The query set to the selected option, which causes the list of options to contain only the selected option after a reset. This only happens if the v-model is a ref that has an initial value. If the v-model is not a ref, or if it is an empty ref, the problem does not occur.

You can see that the result of the options call has only 1 result in this video: https://www.loom.com/share/d426a4deef2b4ccfa6e06c4dce92338a. The network tab isn't visible unfortunately, but a call to this url is made after selecting The Goblet of Fire: https://api.themoviedb.org/3/search/movie?query=Harry%20Potter%20and%20the%20Goblet%20of%20Fire

### Environment

• OS: MacOS
• Browser Chrome

**Comment by sashamilenkovic** [comment]
@mweghorst Hey! If you check out the next tag in the playground (so you're pulling down [0.111.8-4cc3687](https://www.npmjs.com/package/@formkit/pro/v/0.111.8-4cc3687)). Can you confirm that it's working correctly now?

**Comment by sashamilenkovic** [comment]
Also, your issue #656 should be resolved as well.

**Comment by mweghorst** [comment]
> which causes the list of options to contain only the selected option after a reset

This part is indeed fixed in `@next`, because the autocomplete does a fresh call to the options method. However, upon selecting one of the options a call is still made to the options method with the selected value as the query (i.e. to https://api.themoviedb.org/3/search/movie?query=Harry%20Potter%20and%20the%20Goblet%20of%20Fire after clicking on The Goblet of Fire). As a result, a limited set of results (based on the query) is visible for a very short time. I can't seem to upgrade the Playground to Pro@next, so I'm sharing a video of my local dev environment: https://www.loom.com/share/d3690ba106204bc1a57f7ac50aa2678f

> Also, your issue https://github.com/formkit/formkit/issues/656 should be resolved as well.

Yes that issue is resolved, thanks!!

**Comment by sashamilenkovic** [comment]
@mweghorst Gotcha. It makes sense that another request is made to your "optionsLoader" since removing the selected value does "change" the search value to empty string. It would be better if that weren't the case. I can continue taking a look into this.

**Comment by sashamilenkovic** [comment]
@mweghorst Hey sorry for the delay.  I just published a new version of the pro package at the latest tag, https://www.npmjs.com/package/@formkit/pro/v/0.111.9. When you have time, can you take a look and let me know where we stand on your issues?

**Comment by sashamilenkovic** [comment]
@mweghorst One thing to notice, I changed the behavior of open-on-click. Clicking the "removeSElection" button will not open the listbox like it used to (I feel that behavior should be handled by a separate prop maybe in the future).

**Comment by mweghorst** [comment]
> However, if we do not make the initial request when the option is selected, when you reopen the autocomplete, you're going to see the "old" options first

Is this not only an issue when using `selection-appearance="text-input"`? When using `"option"` the autocomplete can only be reopened with an empty query I think.

> you'll always see the old options changed out for new ones

This is exactly what is happening for my input right now but the other way around if you understand what I mean. You'll see the results of query=selectedLabel being replaced by the results of query=null: https://www.loom.com/share/d3690ba106204bc1a57f7ac50aa2678f

**Comment by sashamilenkovic** [comment]
@mweghorst Got it. I'll try to get this implemented by early next week.

**Comment by mweghorst** [comment]
> @mweghorst Got it. I'll try to get this implemented by early next week.

Awesome!

**Comment by sashamilenkovic** [comment]
@mweghorst Hey! Finally got around to working on this: https://formkit.link/519e2b1a81ea636ed41e398a13df6021. I added the "clearOnClick" prop we talked about, do you want to give this a look when you get a chance? https://www.npmjs.com/package/@formkit/pro/v/0.111.13-390d1a3 is the version.

**Comment by mweghorst** [comment]
@sashamilenkovic Sorry I didn't get around to this earlier but I've just tested this and it's working great! There is only one thing I noticed but it's very minor: when typing a query (i.e. Star Wars), selecting an entry, and then clear-on-clicking it the suggestion box shortly shows/flickers Star Wars movies before loading the Harry Potter ones. But again this is very minor and not something I'm even sure could be fixed

Many thanks for this!
